lib/stat_board/graph_helper.rb

```diff
@@ -14,11 +14,27 @@ def first_day_ever
     # along the date_range, on every displayed interval
     def resources_by_date(klass_name)
       klass       = klass_name.to_s.constantize
-      created_ats = klass.pluck(:created_at).compact.map(&:to_date)
-
-      date_range.step(date_steps).map do |date|
-        created_ats.count{|c_a| c_a < date}
-      end.to_s
+      created_ats = klass.order(:created_at).pluck(:created_at).compact
+      steps       = date_range.step(date_steps).map(&:end_of_day)
+      counts      = created_ats.reduce(Hash.new(0)) do |counts, timestamp|
+        cutoff = steps[counts[:index]]
+
+        # As long as timestamps have not exceeded the last date step
+        if cutoff.present?
+          if timestamp < cutoff
+            counts[cutoff] += 1
+          else
+            new_index            = counts[:index] += 1
```

Comment:
keeping your walking index in the counts hash is twisting my brain a bit, cool with moving this logic into another variable?